code-challenges.test.js

```diff
@@ -11,51 +11,148 @@
+    })
+})
 
 // b) Create the function that makes the test pass.
 
-
+//5. create a function "secretCode" then .split("")
+//6. .map() goes through every index looking for a,e,i,o,u (both lowercase and uppercase)
+//7. return a, A to 4, e, E to 3, i, I to 1, o, O to 0
+//8. .join("") change it into a string again
+//9. Run the test yarn jest
+//10. Got green
+
+const secretCode = (string) => {
+    let stringArray = string.split("")
+    let newCodeLet = stringArray.map(value => {
+        if(value === "a" || value === "A"){ return 4 }
+        else if(value === "e" || value === "E") { return 3 }
+        else if(value === "i" || value === "I") { return 1 }
+        else if (value === "o" || value === "O") { return 0 }
+        else { return value }
+    })
+        return newCodeLet.join("")
+  }
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
-
+// const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+// const letterA = "a"
+// // Expected output: ["Apple", "Banana", "Orange"]
+// const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+// const letterE = "e"
+// // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+//1. crete a function "fruitArray" then takes an array of word and single letter and return all the words that contain the particular letter
+//2. expect the input of arrayOfWords1 and letterA to equal ["Apple", "Banana", "Orange"]
+//3. Run yarn jest
+//4. Got red
+
+describe("fruitArray", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+      expect(fruitArray(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+      expect(fruitArray(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+//5. create a function fruitArray has array and letter as parameters
+//6. .filter() filters out only the values that include both lowercase and uppercase
+//7. return the result (newFruitArray)
+//8. Run yarn jest
+//9. Got green
+
+const fruitArray = (array, letter) => {
+    let newFruitArray = array.filter(value => {
+      return value.includes(letter.toLowerCase()) || value.includes(letter.toUpperCase())
+    })
+      return newFruitArray
+}
```

Comment:
Great job. If you find yourself creating a variable just to return it you can just return the logic. `return newFruitArray = array.filter(value.....`